action/candidate_register.go

```diff
@@ -129,6 +146,12 @@ func NewCandidateRegister(
 		}
 		cr.ownerAddress = ownerAddress
 	}
+	// Apply options
+	for _, opt := range options {
+		if err := opt(cr); err != nil {
+			return nil, errors.Wrap(err, "failed to apply option")
+		}
+	}
```

Comment:
we may just have `func SetBLSPublicKey` to avoid the option implementation or `func NewCandidateRegisterWithBLSPublicKey`